### ISSUE TITLE
Add option to jitter outliers in a boxplot

### ISSUE THREAD

**Issue by jolars** [tidyverse/ggplot2#4480]
I would like to be able to add a small amount of jittering to outliers in a boxplot or alternatively stack the points to avoid having them overlap.

Here is an example of where points in a boxplot overlap:

``` r
library(ggplot2)
library(dplyr)

# outliers are overlapping
ggplot(mpg, aes(drv, cty)) +
  geom_boxplot()
```

![](https://i.imgur.com/PzsKZII.png)

To add jittering to these outliers, we currently have to result to the following hack, by creating a separate dataset of outliers and plotting them using `geom_jitter()` manually.

``` r
# adding jittering to outliers is a bit of work
outliers <- 
  mpg %>%
  group_by(drv) %>%
  filter(cty > quantile(cty, 0.75) + 1.5 * IQR(cty) | 
           cty < quantile(cty, 0.25) - 1.5 * IQR(cty))

ggplot(mpg, aes(drv, cty)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height = 0, width = 0.1, data = outliers)
```

![](https://i.imgur.com/66YKeYh.png)

I understand that the `position` argument in `geom_boxplot()` is already "occupied", so maybe the simplest solution would probably to just add a new argument `outlier.jitter = c(0, 0)` (for x and y coordinate jittering respectively).

An even better solution would of course be to incorporate the beeswarm algorithm from **ggbeeswarm**:

``` r

library(ggbeeswarm)

ggplot(mpg, aes(drv, cty)) +
  geom_boxplot(outlier.shape = NA) +
  geom_beeswarm(data = outliers)
```

![](https://i.imgur.com/vfF9qme.png)

<sup>Created on 2021-05-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by teunbrand** [comment]
I don't think that it would make sense to add this as an option to `geom_boxplot()`.
Instead, I think it would make more sense to make a `stat_boxplot_outliers()` that you can use with `geom_point()`, `geom_jitter()` or `ggbeeswarm::geom_beeswarm()` to have positions to your liking.
